addons/sale/models/sale.py

```diff
@@ -1265,3 +1262,9 @@ def _onchange_discount(self):
             discount = (new_list_price - price) / new_list_price * 100
             if discount > 0:
                 self.discount = discount
+
+    def _get_default_name_based_on_product(self, product):
```

Comment:
We need a method on sale order line for event because the descriptio we need to generate needs not only information from the product but fields from the sale order itself.

Reply:
explain also why product is an argument instead of simpy using  self.product